### PR TITLE
[raysgd] Update to fix examples out of the box

### PR DESCRIPTION
**Pull request by maximsmol**
## Why are these changes needed?

The old TF2 SGD implementation is rudimentary. The examples do not run out of the box.

* `tf-example-sgd.yml` has been updated to the latest TF2 version and verified to work with `tensorflow_train_example.py`.
* `tf_trainer.py` documentation has been improved to list the peculiarities of TF2's `MultiWorkerMirroredStrategy`, which behaves very differently from distributed learning in PyTorch.
* `tensorflow_train_example.py` now runs on the GPU by default and prints losses before and after training to verify that the model converges.

## Related issue number

N/A

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.

### CODE REVIEW

**Comment by richardliaw** [comment]
Hi Maxim, https://travis-ci.com/ray-project/ray/jobs/281740528 is failing.

**Comment by richardliaw** [comment]
jenkins test tune

File: python/ray/experimental/sgd/tf/examples/tensorflow_train_example.py [hunk 1]
```diff
@@ -68,16 +69,14 @@ def train_example(num_replicas=1, batch_size=128, use_gpu=False):
...
-    train_stats2 = trainer.train()
-    train_stats2.update(trainer.validate())
-    print(train_stats2)
+    trainer.train()
+    trainer.train()
 
-    val_stats = trainer.validate()
-    print(val_stats)
+    end_stats = trainer.validate()
+    print(end_stats)
```
Comment (richardliaw) [inline python/ray/experimental/sgd/tf/examples/tensorflow_train_example.py]: What are these changes for? Can you leave a comment?

File: python/ray/experimental/sgd/tf/examples/tf-example-sgd.yaml [hunk 2]
```diff
@@ -47,13 +47,16 @@ worker_nodes:
...
+    - conda install setuptools=45.1.0=py36_0 wrapt=1.11.2 --yes # workaround to fix wrapt error
+    - ray &> /dev/null || pip install -U https://s3-us-west-2.amazonaws.com/ray-wheels/latest/ray-0.9.0.dev0-cp36-cp36m-manylinux1_x86_64.whl
+    - pip install -U ray[tune]
+    - pip install tensorflow==2.1.0
+    # todo: git clone distribution token
+
+file_mounts:
+    "~/anaconda3/lib/python3.6/site-packages/ray/experimental/sgd/pytorch/": "../../pytorch/"
+    "~/anaconda3/lib/python3.6/site-packages/ray/experimental/sgd/tf/": "../../tf/"
+    "~/anaconda3/lib/python3.6/site-packages/ray/experimental/sgd/examples/": "."
```
Comment (richardliaw) [inline python/ray/experimental/sgd/tf/examples/tf-example-sgd.yaml]: Don't forget to revert these changes for filemounts on site-packages

**Comment by richardliaw** [comment]
BTW, CI is failing for `tensorflow_train_example.py`: https://amplab.cs.berkeley.edu/jenkins/job/Ray-PRB/21157/console

File: python/ray/experimental/sgd/tf/examples/tf-example-sgd.yaml (hunk 2 above)
Comment (maximsmol) [reply python/ray/experimental/sgd/tf/examples/tf-example-sgd.yaml]: I removed them in a previous commit and pushed them back in by accident, my bad

**Comment by maximsmol** [comment]
jenkins test tune

**Review by richardliaw — APPROVED** [review]
